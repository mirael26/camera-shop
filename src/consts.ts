export const AppUrl = {
  Main: '/',
  Catalog: '/catalog',
  Page: '/page_',
  Product: '/product',
  Cart: '/cart',
  NotFound: '/404',
  UnknownError: '/unknown_error',
  ServerUnavailable: '/server_unavailable',
} as const;

export const ApiUrl = {
  Promo: '/promo',
  Products: '/cameras',
  Reviews: '/reviews',
  Similar: '/similar',
  Promocode: '/coupons',
  Order: '/orders',
} as const;

export const Modal = {
  AddToCart: 'addToCartModal',
  DeleteFromCart: 'deleteFromCartModal',
  Review: 'reviewModal',
  OrderSuccess: 'orderSuccessModal',
} as const;

export const Tab = {
  Features: 'features',
  Description: 'description',
} as const;

export const Month = {
  0: 'января',
  1: 'февраля',
  2: 'марта',
  3: 'апреля',
  4: 'мая',
  5: 'июня',
  6: 'июля',
  7: 'августа',
  8: 'сентября',
  9: 'октября',
  10: 'ноября',
  11: 'декабря',
} as const;

export const Param = {
  Sort: 'sort',
  Order: 'order',
  Page: 'page',
  Tab: 'tab',
  PriceMin: 'price_min',
  PriceMax: 'price_max',
  Category: 'category',
  Level: 'level',
  Type: 'type',
} as const;

export const SortType = {
  Price: 'price',
  Rating: 'rating',
} as const;

export const SortOrder = {
  Asc: 'asc',
  Desc: 'desc',
} as const;

export const Category = {
  Camera: 'Фотоаппарат',
  VideoCamera: 'Видеокамера',
} as const;

export const Level = {
  Beginner: 'Нулевой',
  Amateur: 'Любительский',
  Professional: 'Профессиональный',
} as const;

export const CameraType = {
  Digital: 'Цифровая',
  Film: 'Плёночная',
  Instant: 'Моментальная',
  Collectible: 'Коллекционная',
} as const;
