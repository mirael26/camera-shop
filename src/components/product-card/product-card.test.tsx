import { fireEvent, getByText, screen } from '@testing-library/react';
import { AppUrl } from '../../consts';
import { renderTestApp } from '../../test/helpers/render-test-app';
import { renderWithReduxAndRouter } from '../../test/helpers/render-with-redux-and-router';
import { productMock } from '../../test/mocks';
import ProductCard from './product-card';

describe('ProductCard', () => {
  test('Render correctly', () => {
    const product = renderWithReduxAndRouter(<ProductCard product={productMock}/>);
    expect(product).toMatchSnapshot();
  });

  test('Display product data correctly', () => {
    renderWithReduxAndRouter(<ProductCard product={productMock}/>);
    expect(screen.getByTestId('rating')).toHaveTextContent('3');
    expect(screen.getByTestId('rate-count')).toHaveTextContent('17');
    expect(screen.getByTestId('name')).toHaveTextContent(/Орлёнок/i);
    expect(screen.getByTestId('price')).toHaveTextContent(/19 970/i);
  });

  test('Click on buy-button opens popup', () => {
    renderTestApp(<ProductCard product={productMock}/>, { route: AppUrl.Catalog });
    const productCard = screen.getByTestId('product-card');
    const buyButton = getByText(productCard, /Купить/i);
    fireEvent.click(buyButton);
    expect(screen.getByTestId('add-to-cart-modal')).toBeInTheDocument();
  });

  test('Click on details-link redirect to product page', () => {
    renderTestApp(<ProductCard product={productMock}/>, { route: AppUrl.Catalog });
    const productCard = screen.getByTestId('product-card');
    const detailsLink = getByText(productCard, /Подробнее/i);
    fireEvent.click(detailsLink);
    expect(screen.getByTestId('product-page')).toBeInTheDocument();
  });
});
