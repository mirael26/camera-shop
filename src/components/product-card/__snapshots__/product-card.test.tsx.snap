// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="product-card"
        data-testid="product-card"
      >
        <div
          class="product-card__img"
        >
          <picture>
            <source
              srcset="img/content/orlenok.webp, img/content/orlenok@2x.webp 2x"
              type="image/webp"
            />
            <img
              alt="Орлёнок"
              height="240"
              src="img/content/orlenok.jpg"
              srcset="img/content/orlenok@2x.jpg 2x"
              width="280"
            />
          </picture>
        </div>
        <div
          class="product-card__info"
        >
          <div
            class="rate product-card__rate"
          >
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <p
              class="visually-hidden"
              data-testid="rating"
            >
              Рейтинг: 
              3
            </p>
            <p
              class="rate__count"
              data-testid="rate-count"
            >
              <span
                class="visually-hidden"
              >
                Всего оценок:
              </span>
              17
            </p>
          </div>
          <p
            class="product-card__title"
            data-testid="name"
          >
            Орлёнок
          </p>
          <p
            class="product-card__price"
            data-testid="price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            19 970
             ₽
          </p>
        </div>
        <div
          class="product-card__buttons"
        >
          <button
            class="btn btn--purple product-card__btn"
            type="button"
          >
            Купить
          </button>
          <a
            class="btn btn--transparent"
            href="/catalog/product/4"
          >
            Подробнее
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="product-card"
      data-testid="product-card"
    >
      <div
        class="product-card__img"
      >
        <picture>
          <source
            srcset="img/content/orlenok.webp, img/content/orlenok@2x.webp 2x"
            type="image/webp"
          />
          <img
            alt="Орлёнок"
            height="240"
            src="img/content/orlenok.jpg"
            srcset="img/content/orlenok@2x.jpg 2x"
            width="280"
          />
        </picture>
      </div>
      <div
        class="product-card__info"
      >
        <div
          class="rate product-card__rate"
        >
          <svg
            aria-hidden="true"
            height="16"
            width="17"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="16"
            width="17"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="16"
            width="17"
          >
            <use
              xlink:href="#icon-full-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="16"
            width="17"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
          <svg
            aria-hidden="true"
            height="16"
            width="17"
          >
            <use
              xlink:href="#icon-star"
            />
          </svg>
          <p
            class="visually-hidden"
            data-testid="rating"
          >
            Рейтинг: 
            3
          </p>
          <p
            class="rate__count"
            data-testid="rate-count"
          >
            <span
              class="visually-hidden"
            >
              Всего оценок:
            </span>
            17
          </p>
        </div>
        <p
          class="product-card__title"
          data-testid="name"
        >
          Орлёнок
        </p>
        <p
          class="product-card__price"
          data-testid="price"
        >
          <span
            class="visually-hidden"
          >
            Цена:
          </span>
          19 970
           ₽
        </p>
      </div>
      <div
        class="product-card__buttons"
      >
        <button
          class="btn btn--purple product-card__btn"
          type="button"
        >
          Купить
        </button>
        <a
          class="btn btn--transparent"
          href="/catalog/product/4"
        >
          Подробнее
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
